CMakeLists.txt

```diff
@@ -252,15 +238,23 @@ elseif(CMAKE_C_COMPILER_ID MATCHES "GNU" OR CMAKE_C_COMPILER_ID MATCHES "Clang")
     if(MINGW)
         list(APPEND WARNFLAGS_DISABLE -Wno-pedantic-ms-format)
     endif()
-else()
-    if(WITH_NATIVE_INSTRUCTIONS)
+endif()
+
+if(WITH_NATIVE_INSTRUCTIONS)
+    # Set native march/mcpu
+    if(CMAKE_C_COMPILER_ID MATCHES "GNU" OR CMAKE_C_COMPILER_ID MATCHES "Clang")
+        if(BASEARCH_PPC_FOUND)
+            set(NATIVEFLAG "-mcpu=native")
+        else()
+            set(NATIVEFLAG "-march=native")
+        endif()
+        add_compile_options(${NATIVEFLAG})
+    else()
         message(STATUS "Ignoring WITH_NATIVE_INSTRUCTIONS; not implemented yet on this configuration")
         set(WITH_NATIVE_INSTRUCTIONS OFF)
     endif()
-endif()
-
-# Disable LTO
-if(NOT WITH_NATIVE_INSTRUCTIONS)
+else()
+    # Disable LTO
```

Comment:
``WITH_NATIVE_INSTRUCTIONS`` can be turned off in the previous ``if`` if compiler does not support a native flag.
I think we need to check the ``NOT  WITH_NATIVE_INSTRUCTIONS`` condition again.

Reply:
Technically, you are correct.
But, if you ASK for native, then you should not also ask for LTO since we don't support that combination.
I don't really think it makes sense to allow LTO anyway in the case that native failed to activate, that would be less predictable than just rejecting it like advertised.

Reply:
This code doesn't enable LTO, it disable LTO.
If we can't activate native compilation we need to disable LTO.

Reply:
Sorry, I think I confused myself (and you possibly) yesterday.

Legal possibilities:
- No LTO, No native
- No LTO, Native
- LTO, Native

That leaves us with this rule: If you don't ask for native, we always disable LTO.

So that means you are right, there is a bug here; we should disable LTO if native fails to activate.
Thanks for sticking with it 👍

Reply:
@phprus Fixed, please verify.

Reply:
@Dead2 
LGTM